kernel/src/sched/sched_class/idle.rs

```diff
@@ -35,11 +35,8 @@ impl core::fmt::Debug for IdleClassRq {
 
 impl SchedClassRq for IdleClassRq {
     fn enqueue(&mut self, entity: Arc<Task>, _: Option<EnqueueFlags>) {
-        let ptr = Arc::as_ptr(&entity);
-        if let Some(t) = self.entity.replace(entity)
-            && ptr != Arc::as_ptr(&t)
-        {
-            panic!("Multiple `idle` entities spawned")
+        if self.entity.replace(entity).is_some() {
```

Comment:
This is `assert!(self.entity.replace(entity).is_none(), <msg>)`

Reply:
Let's replace `assert` with `debug_assert` (or `if` plus `panic`) since this property should have been guaranteed by the sole user of `IdleClassRq`, the `sched_class/mod.rs` file. We should not have to check it over and over again in production.

```rust
let old = self.entity.replace(entity);
debug_assert!(old.is_none(), "the length of the idle queue should be no larger than 1");
```